synapse/storage/databases/main/event_push_actions.py

```diff
@@ -458,57 +473,151 @@ def _get_thread(thread_id: str) -> NotifCounts:
         # updated `event_push_summary` synchronously when persisting a new read
         # receipt).
         txn.execute(
-            """
-                SELECT stream_ordering, notif_count, COALESCE(unread_count, 0), thread_id
+            f"""
+                SELECT notif_count, COALESCE(unread_count, 0), thread_id
                 FROM event_push_summary
+                LEFT JOIN (
+                    SELECT thread_id, MAX(stream_ordering) AS receipt_stream_ordering
+                    FROM receipts_linearized
+                    LEFT JOIN events USING (room_id, event_id)
+                    WHERE
+                        user_id = ?
+                        AND room_id = ?
+                        AND {receipt_types_clause}
+                    GROUP BY thread_id
+                ) AS receipts USING (thread_id)
```

Comment:
I think this can be summarized as "Get the max `receipt_stream_ordering` of threads in the room for the user".

```suggestion
                LEFT JOIN (
                    /* Get the max `receipt_stream_ordering` of the threaded world in the room for the user
                     * because TODO: why
                     */
                    SELECT thread_id, MAX(stream_ordering) AS receipt_stream_ordering
                    FROM receipts_linearized
                    LEFT JOIN events USING (room_id, event_id)
                    WHERE
                        user_id = ?
                        AND room_id = ?
                        AND {receipt_types_clause}
                    GROUP BY thread_id
                ) AS receipts USING (thread_id)
```

Reply:
The inner query is getting the latest receipt per thread, yep. 👍  I think this is now clear from the comments at the top of the function.